use-cases/school-timetabling/src/main/resources/META-INF/resources/app.js

```diff
@@ -218,6 +222,36 @@ function solve() {
   "text");
 }
 
+function analyze() {
+  new bootstrap.Modal("#scoreAnalysisModal").show()
```

Comment:
Consider more friendly message in case the data set has not been solved yet. As of now, the message starts with "{"details":"Error id c9e674b0-daca-4829-8db8-326753a24402-2, org.jboss.resteasy.spi.UnhandledException ..." and only later, if one  reads carefully, they learn what the root cause is.

For unexpected exceptions, it's ok. In this case, I would argues it's not, as halve the users will be tempted to click the score analysis button first.